### PR TITLE
Added `PrepareProposal` and `ProcessProposal` data `CHANGELOG_PENDING.md`. Reworked `UPGRADING.md`

### LINKED ISSUE #9304: Rework of `CHANGELOG_PENDING.md` and `UPGRADING.md` to reflect the code merged from `feature/abci++ppp`
Most PRs in `feature/abci++ppp` did not update `CHANGELOG_PENDING.md`, so it could be done in a holistic manner when the work was complete.

This issue's goal is a rework of `CHANGELOG_PENDING.md` to sync it with the changes coming from `feature/abci++ppp`.

As part if this issue file `UPGRADING.md` will also be reworked.

### PR DESCRIPTION
**Pull request by sergio-mena**
Closes #9304 

This is a documentation PR.

* `CHANGELOG_PENDING.md` now contains entries reflecting the changes coming from `feature/abci++ppp`
* `UPGRADING.md` has been compared to its version in `v0.35.x`, `v0.36.x` and `master` and all relevant differences have been included
  * In particular #5404 has been cherry-picked

---

#### PR checklist

- [x] Tests written/updated, or no tests needed
- [x] `CHANGELOG_PENDING.md` updated, or no changelog entry needed
- [x] Updated relevant documentation (`docs/`) and code comments, or no
      documentation updates needed

### CODE REVIEW

File: CHANGELOG_PENDING.md [hunk 1]
```diff
@@ -9,31 +9,31 @@ Friendly reminder, we have a [bug bounty program](https://hackerone.com/tendermi
...
   - [abci] \#5783 Make length delimiter encoding consistent (`uint64`) between ABCI and P2P wire-level protocols
   - [abci] \#9145 Removes unused Response/Request `SetOption` from ABCI (@samricotta)
-  - [abci] \#8656 Added cli command for `PrepareProposal`. (@jmalicevic)
-  - [abci] \#8901 Added cli command for `ProcessProposal`. (@hvanz)
   - [abci/params] \#9287 Deduplicate `ConsensusParams` and `BlockParams` so only `types` proto definitions are used (@cmwaters)
     - Remove `TimeIotaMs` and use a hard-coded 1 millisecond value to ensure monotonically increasing block times.
     - Rename `AppVersion` to `App` so as to not stutter.
+  - [abci] \#9301 New ABCI methods `PrepareProposal` and `ProcessProposal` which give the app control over transactions proposed and allows for verification of proposed blocks.
+  - [abci] \#8656 Added cli command for `PrepareProposal`. (@jmalicevic)
+  - [abci] \#8901 Added cli command for `ProcessProposal`. (@hvanz)
```
Comment (tychoish) [inline CHANGELOG_PENDING.md:23]: I'd combine these lines.

**Review by tychoish — APPROVED** [review]

File: CHANGELOG_PENDING.md (hunk 1 above)
Comment (sergio-mena) [reply CHANGELOG_PENDING.md:23]: Yes, good idea. Is it OK (in terms of current practice) to then "list" the PRs and the authors? Something like:
```md
- [abci] \#8656, \#8901 Added cli commands for `PrepareProposal`, `ProcessProposal` (@jmalicevic, @hvanz) 
```
My concern is that is might break the script that turns PR IDs into links...

Comment (thanethomson) [reply CHANGELOG_PENDING.md:23]: The script does seem to work on this (I tested it out locally to figure out how it works) - it's a very simple find/replace script.

File: UPGRADING.md [hunk 2]
```diff
@@ -7,11 +7,25 @@ Tendermint Core.
...
+* The `ABCIVersion` is now `0.18.0`.
+
+* Added new ABCI methods `PrepareProposal` and `ProcessProposal`. For details,
+  please see the [spec](spec/abci/README.md). Applications upgrading to
+  v0.37.0 must implement these methods, at the very minimum, as described
+  [here](spec/abci/apps.md)
+* Deduplicated `ConsensusParams` and `BlockParams`.
+  In the v0.34 branch they are defined both in `abci/types.proto` and `types/params.proto`.
+  The definitions in `abci/types.proto` have been removed.
+  This in process applications should make sure they are not using the deleted
```
Comment (thanethomson) [inline UPGRADING.md:19]: Did you perhaps mean something like:

```suggestion
  In-process applications should make sure they are not using the deleted
```

File: UPGRADING.md [hunk 3]
```diff
@@ -7,11 +7,25 @@ Tendermint Core.
...
+  version of those structures.
 * In v0.34, messages on the wire used to be length-delimited with `int64` varint
   values, which was inconsistent with the `uint64` varint length delimiters used
   in the P2P layer. Both now consistently use `uint64` varint length delimiters.
-* Added `AbciVersion` to `RequestInfo`. Applications should check that the ABCI
-  version they expect is being used in order to ensure compatibility.
+* Added `AbciVersion` to `RequestInfo`.
+  Applications should check that Tendermint's ABCI version matces the one they expect
```
Comment (thanethomson) [inline UPGRADING.md:25]:
```suggestion
  Applications should check that Tendermint's ABCI version matches the one they expect
```

File: UPGRADING.md [hunk 4]
```diff
@@ -7,11 +7,25 @@ Tendermint Core.
...
+  in order to ensure compatibility.
+* The method `SetOption` has been removed from the ABCI.Client interface.
+  This feature was used in the early ABCI implementation's.
```
Comment (thanethomson) [inline UPGRADING.md:28]: Should we not also mention something about how we've deprecated this in the protos, like:

```suggestion
* The `SetOption` method has been removed from the ABCI `Client` interface.
  The corresponding Protobuf types have been deprecated.
```

File: UPGRADING.md [hunk 5]
```diff
@@ -75,12 +89,9 @@ directory. For more, see "Protobuf," below.
...
 
-* `Header#LastResultsHash` previously was the root hash of a Merkle tree built from `ResponseDeliverTx(Code, Data)` responses.
-  As of 0.34,`Header#LastResultsHash` is now the root hash of a Merkle tree built from:
-    * `BeginBlock#Events`
-    * Root hash of a Merkle tree built from `ResponseDeliverTx(Code, Data,
-      GasWanted, GasUsed, Events)` responses
-    * `BeginBlock#Events`
+* `Header#LastResultsHash`, which is the root hash of a Merkle tree built from
+`ResponseDeliverTx(Code, Data)` as of v0.34 also includes `GasWanted` and `GasUsed`
+fields.
```
Comment (thanethomson) [inline UPGRADING.md:94]:
```suggestion
* `Header#LastResultsHash`, which is the root hash of a Merkle tree built from
  `ResponseDeliverTx(Code, Data)` as of v0.34 also includes `GasWanted` and `GasUsed`
  fields.
```

File: UPGRADING.md [hunk 6]
```diff
@@ -184,6 +195,7 @@ Other user-relevant changes include:
 * The `Verifier` was broken up into two pieces:
     * Core verification logic (pure `VerifyX` functions)
     * `Client` object, which represents the complete light client
+* The new light clients stores headers & validator sets as `LightBlock`s
```
Comment (thanethomson) [inline UPGRADING.md:198]:
```suggestion
* The new light client stores headers and validator sets as `LightBlock`s
```

**Review by thanethomson — APPROVED** [review]
Just some minor nits, but otherwise looks great!

**Pull request merged**